runtime/common/src/parachains.rs

```diff
@@ -228,7 +240,7 @@ pub trait Trait: attestations::Trait + session::historical::Trait {
 	/// Proof type (since it is a part of Calls it needs to be bound to `Parameter` here)
 	/// TODO: I know of no other way to bind an associated type of `KeyOwnerProofSystem::Proof`
```

Comment:
you can have `where` clauses on associated types, AFAIK

Reply:
But the `type KeyOwnerProofSystem<Proof = Self::Proof>` is a normal way to do it as well. `IntoIterator` is defined using the same trick

Reply:
Using `where` on associated types is unstable, this would fail:

```rust
trait A {
    type C where Self::C: Copy;
}
```